exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSystemPlugin.java

```diff
@@ -104,6 +109,33 @@ public FileSystemPlugin(FileSystemConfig config, DrillbitContext context, String
     }
   }
 
+  private boolean isS3Connection(Configuration conf) {
+    URI uri = FileSystem.getDefaultUri(conf);
+    return uri.getScheme().equals("s3a");
+  }
+
+  /**
+   * Retrieve secret and access keys from configured (with
+   * {@link org.apache.hadoop.security.alias.CredentialProviderFactory#CREDENTIAL_PROVIDER_PATH} property)
+   * credential providers and set it into {@code conf}. If provider path is not configured or credential
+   * is absent in providers, it will conditionally fallback to configuration setting. The fallback will occur unless
+   * {@link org.apache.hadoop.security.alias.CredentialProvider#CLEAR_TEXT_FALLBACK} is set to {@code false}.
+   *
+   * @param conf {@code Configuration} which will be updated with credentials from provider
+   * @throws IOException thrown if a credential cannot be retrieved from provider
+   */
+  private void handleS3Credentials(Configuration conf) throws IOException {
+    String[] credentialKeys = {"fs.s3a.secret.key", "fs.s3a.access.key"};
```

Comment:
Consider using org.apache.hadoop.fs.s3a.Contants class.

Reply:
I am aware of the Constants, but artifact (hadoop-aws), containing this class is not among the module's dependencies (however it is present in distribution's (compile-scope) and drill-root's (test-scope) dependencies). Is there a need to add the dependency?